Sim_To_DuT_Interface/DuTLogger/DuTLogger.cpp

```diff
@@ -0,0 +1,339 @@
+    return newHandler;
+}
+
+/**
+ * Creates a handler to write messages in a file.
+ * This handler can be used to connect it to a logger.
+ * It will create a file for this day, if it hasn't been done yet.
+ *
+ * @return a hadler to log messages in a file
+ */
+quill::Handler* DuTLogger::buildFileHandler() {
+    // a second handler for the file is needed
+    std::string basicPath = currentLogpathConsole + "/Logfile.log";
+    quill::Handler* newHandler = quill::file_handler(basicPath, FILE_MODE_CONSOLE,
+                                                     quill::FilenameAppend::Date);
+    newHandler->set_log_level(DEFAULT_FILE_LOG_LEVEL);
+
+    // modify the pattern for the logger
+    newHandler->set_pattern(QUILL_STRING("%(ascii_time)  %(level_name): %(message)"),
+                            "%D %H:%M:%S.%Qms");
+    // return the new handler
+    return newHandler;
+}
+
+/**
+ * Builds a logger for the console. If necessary the logger will be build with an additional handler for
+ * files. This will be configured by the second argument.
+ *
+ * @param name the name for the logger
+ * @param withFileHandler true if the logger also needs a handler for logfiles
+ * @return the created logger
+ */
+quill::Logger* DuTLogger::createConsoleLogger(const char* name, bool withFileHandler) {
+    // check if the quill engine is started
+    startEngine();
+
+    // check if the new logger needs to write the console log to a file
+    // if it needs it, we have to build the logger a different way
+    quill::Logger* newLogger;
+    if (withFileHandler) {
+        newLogger = quill::create_logger(name, {consoleHandler, consoleFileHandler});
+
+        // check if we might have to delete old logging folders
+        removeOldLogfiles(currentLogpathConsole);
+    } else {
+        newLogger = quill::create_logger(name, consoleHandler);
+    }
+
+    // Set the LogLevel (L3 for everything)
+    newLogger->set_log_level(quill::LogLevel::TraceL3);
+
+    return newLogger;
+}
+
+/**
+ * Creates a logger to log a specific data object (like events; not normal messages!) into a file.
+ * The logger will be build with all necessary handlers.
+ * It will create a file for this day, if it hasn't been done yet.
+ *
+ * @return a logger to log data objects into a file
+ */
+quill::Logger* DuTLogger::createDataLogger() {
+    // create a file handler to connect quill to a logfile
+    std::string basicPath = currentLogpathData + "/Logfile.log";
+    quill::Handler* file_handler = quill::file_handler(basicPath, FILE_MODE_DATA,quill::FilenameAppend::Date);
+
+    // configure the pattern of a line
+    file_handler->set_pattern(QUILL_STRING("%(ascii_time) %(logger_name) - %(message)"),
+            "%D %H:%M:%S.%Qms");
+
+    // finally, create the logger and return it
+    quill::Logger* createdLogger = quill::create_logger("dataLog", file_handler);
+
+    // check if we might have to delete old logging folders
+    removeOldLogfiles(currentLogpathData);
```

Comment:
this should not happen, we dont want to accidentally delete old files by rerunning the simulation

Reply:
Backup Count is a nice solution for that, nvm. but create a new file for every run